tests/qos/test_qos_sai.py

```diff
@@ -181,6 +181,63 @@ def testQosSaiHeadroomPoolSize(self, ptfhost, dutTestParams, dutConfig, dutQosCo
         testParams.update(dutTestParams["basicParams"])
         self.runPtfTest(ptfhost, testCase="sai_qos_tests.HdrmPoolSizeTest", testParams=testParams)
 
+    @pytest.mark.parametrize("bufPool", ["wm_buf_pool_lossless", "wm_buf_pool_lossy"])
+    def testQosSaiBufferPoolWatermark(self, request, bufPool, ptfhost, dutTestParams, dutConfig, dutQosConfig, ingressLosslessProfile, egressLossyProfile, resetWatermark):
+        """
+            Test QoS SAI Queue buffer pool watermark for lossless/lossy traffic
+
+            Args:
+                ptfhost (AnsibleHost): Packet Test Framework (PTF)
+                dutTestParams (Fixture, dict): DUT host test params
+                dutConfig (Fixture, dict): Map of DUT config containing dut interfaces, test port IDs, test port IPs,
+                    and test ports
+                dutQosConfig (Fixture, dict): Map containing DUT host QoS configuration
+                ingressLosslessProfile (Fixture): Map of ingress lossless buffer profile attributes
+                egressLossyProfile (Fixture): Map of egress lossy buffer profile attributes
+                resetWatermark (Fixture): reset watermarks
+
+            Returns:
+                None
+
+            Raises:
+                RunAnsibleModuleFail if ptf test fails
+        """
+        disableTest = request.config.getoption("--disable_test")
+        if disableTest:
+            pytest.skip("Buffer Pool watermark test is disabled")
```

Comment:
Is the option necessary or can we rely on the generic mechanism like `run_tests` is using? https://github.com/Azure/sonic-mgmt/blob/5adf0962d15ab5de1548ee59e3b63a81310d980c/tests/run_tests.sh#L108

Reply:
I thought we can use this mechanism only to skip scripts? Can we do it at a per testcase level?

Reply:
I haven't tried it in this specific case but I think you can do something like:
```
--ignore test_qos_sai.py::TestQosSai::testQosSaiBufferPoolWatermark
```

Reply:
Leaving it as is for the time being